### PR TITLE
Update samples to new csproj files

### PR DESCRIPTION
**Pull request by nicolaspierre1990**

### CODE REVIEW

File: src/Beacon/BeaconDemo/App.config [hunk 1]
```diff
@@ -1,6 +1,6 @@
-﻿<?xml version="1.0" encoding="utf-8" ?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/App.config]: All these `App.config` files can be deleted.

File: src/Beacon/BeaconDemo/BeaconDemo.csproj [hunk 2]
```diff
@@ -1,113 +1,29 @@
...
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{D0B707E4-5563-41FA-A8B0-292605332EF1}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>BeaconDemo</RootNamespace>
-    <AssemblyName>BeaconDemo</AssemblyName>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/BeaconDemo.csproj]: I don't think we need binding redirects. Perhaps remove these.

File: src/Beacon/BeaconDemo/BeaconDemo.csproj [hunk 3]
```diff
@@ -1,113 +1,29 @@
...
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{D0B707E4-5563-41FA-A8B0-292605332EF1}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>BeaconDemo</RootNamespace>
-    <AssemblyName>BeaconDemo</AssemblyName>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/BeaconDemo.csproj]: Perhaps move the target framework to a `Directory.Build.props` file in the `src` folder.

File: src/Beacon/BeaconDemo/BeaconDemo.csproj [hunk 4]
```diff
@@ -1,113 +1,29 @@
...
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <AssemblyTitle>BeaconDemo</AssemblyTitle>
+    <Product>BeaconDemo</Product>
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/BeaconDemo.csproj]: These are redundant, as they default to the name of the project file (which is identical).

File: src/Beacon/BeaconDemo/BeaconDemo.csproj [hunk 5]
```diff
@@ -1,113 +1,29 @@
...
+    <Copyright>Copyright ©  2016</Copyright>
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/BeaconDemo.csproj]: I would remove these `Copyright` properties, personally (unless they have real values in them).

File: src/Beacon/BeaconDemo/BeaconDemo.csproj [hunk 6]
```diff
@@ -1,113 +1,29 @@
...
   <ItemGroup>
-    <Compile Include="Bus.cs" />
-    <Compile Include="Program.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Remove="Backup1\**" />
+    <Compile Remove="Backup\**" />
+    <EmbeddedResource Remove="Backup1\**" />
+    <EmbeddedResource Remove="Backup\**" />
+    <None Remove="Backup1\**" />
+    <None Remove="Backup\**" />
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/BeaconDemo.csproj]: These `Remove`d items are not needed.

File: src/Beacon/BeaconDemo/BeaconDemo.csproj [hunk 7]
```diff
@@ -1,113 +1,29 @@
...
   </ItemGroup>
   <ItemGroup>
-    <None Include="App.config" />
-    <None Include="packages.config" />
+    <PackageReference Include="AsyncIO" Version="0.1.69" />
+    <PackageReference Include="NaCl.Net" Version="0.1.13" />
+    <PackageReference Include="NetMQ" Version="4.0.1.8" />
+    <PackageReference Include="System.Memory" Version="4.5.5" />
+    <PackageReference Include="System.Numerics.Vectors" Version="4.5.0" />
+    <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/BeaconDemo.csproj]: It would be handy to specify package versions centrally. This can be done in a `Directory.Build.targets` file.

In `.csproj` files you'd omit the `Version`. i.e.:

```xml
<PackageReference Include="AsyncIO" />
```

Then in that `.targets` file you'd have:

```xml
<PackageReference Update="AsyncIO" Version="0.1.69" />
...
```

File: src/Beacon/BeaconDemo/Bus.cs [hunk 8]
```diff
@@ -1,9 +1,9 @@
-﻿using System;
+﻿using NetMQ;
+using NetMQ.Sockets;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Net;
-using NetMQ;
-using NetMQ.Sockets;
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/Bus.cs]: It's easier to review a huge PR like this if you keep unrelated refactorings separate.

File: src/Majordomo/MDPBrokerProcess/MDPBrokerProcess.csproj [hunk 9]
```diff
@@ -1,122 +1,33 @@
...
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>MDPBrokerProcess</RootNamespace>
-    <AssemblyName>MDPBrokerProcess</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
+    <AssemblyTitle>MDPBrokerProcess</AssemblyTitle>
+    <Product>MDPBrokerProcess</Product>
+    <AssemblyVersion>1.0.15052.1</AssemblyVersion>
+    <FileVersion>1.0.15052.1</FileVersion>
```
Comment (drewnoakes) [inline src/Majordomo/MDPBrokerProcess/MDPBrokerProcess.csproj]: I don't think we need versions on these.

File: src/Majordomo/MDPClientAsyncExample/MDPClientAsyncExample.csproj [hunk 10]
```diff
@@ -1,80 +1,34 @@
...
-    <AssemblyName>MDPClientAsyncExample</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <TargetFrameworkProfile />
+    <AssemblyTitle>MDPClientAsyncExample</AssemblyTitle>
+    <Product>MDPClientAsyncExample</Product>
+    <Copyright>Copyright ©  2016</Copyright>
+    <OutputPath>bin\$(Configuration)\</OutputPath>
```
Comment (drewnoakes) [inline src/Majordomo/MDPClientAsyncExample/MDPClientAsyncExample.csproj]: I wouldn't override `OutputPath` here (or elsewhere) unless there's a good reason to do so. It shouldn't be necessary.

File: src/Majordomo/MDPClientExample/MDPClientExample.csproj [hunk 11]
```diff
@@ -1,16 +1,7 @@
...
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{BA3B66DE-85AB-44F0-890B-E598C777880D}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>MDPClientExample</RootNamespace>
-    <AssemblyName>MDPClientExample</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <TargetFramework>net48</TargetFramework>
     <PublishUrl>publish\</PublishUrl>
```
Comment (drewnoakes) [inline src/Majordomo/MDPClientExample/MDPClientExample.csproj]: Looks like all properties from `<PublishUrl>` onwards can be removed.

File: src/Majordomo/MDPClientExample/MDPClientExampleProgram.cs [hunk 12]
```diff
@@ -63,7 +63,7 @@ private static void Main(string[] args)
                 // create MDP client and set verboseness && use automatic disposal
                 using (var session = new MDPClient("tcp://localhost:5555", id))
                 {
-                    if (verbose)
+                    //if (verbose)
```
Comment (drewnoakes) [inline src/Majordomo/MDPClientExample/MDPClientExampleProgram.cs]: Why this change?

File: src/Majordomo/MDPWorkerExample/MDPWorkerExampleProgram.cs [hunk 13]
```diff
@@ -37,7 +37,7 @@ private static void Main (string[] args)
                 {
                     session.HeartbeatDelay = TimeSpan.FromMilliseconds (10000);
                     // logging info to be displayed on screen
-                    if (verbose)
+                    //if (verbose)
```
Comment (drewnoakes) [inline src/Majordomo/MDPWorkerExample/MDPWorkerExampleProgram.cs]: Ditto.

File: src/Majordomo/MajordomoProtocolTests/MDPBrokerTests.cs [hunk 14]
```diff
@@ -105,7 +105,7 @@ public async void Run_ReceiveREADYMessageFromWorker_LogSuccessfulRegistration ()
                 cts.Cancel ();
                 // check on the logging
                 Assert.That (log.Count, Is.EqualTo (2));
-                Assert.That (log[1], Is.StringContaining ("added to service echo"));
+                Assert.That (log[1], Does.Contain("added to service echo"));
```
Comment (drewnoakes) [inline src/Majordomo/MajordomoProtocolTests/MDPBrokerTests.cs]: Why this change?

File: src/Pirate Pattern/Lazy Pirate/LazyPirate.Client2/Program.cs [hunk 15]
```diff
@@ -44,25 +44,25 @@ private static void SubscriberContinuousLoop(string pubSubAddress, string reques
...
+                        //        break;
+                        //    case RequestSocket.ProgressTopic.Failure:
+                        //        Console.WriteLine("C: Server seems to be offline, abandoning");
+                        //        break;
+                        //    case RequestSocket.ProgressTopic.Success:
+                        //        Console.WriteLine("C: Server replied OK");
+                        //        break;
+                        //    default:
+                        //        throw new ArgumentOutOfRangeException();
+                        //}
```
Comment (drewnoakes) [inline src/Pirate Pattern/Lazy Pirate/LazyPirate.Client2/Program.cs]: Why this change?

File: src/Samples.sln [hunk 16]
```diff
@@ -1,63 +1,63 @@
...
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "HelloWorld", "HelloWorld", "{5A526E6B-083E-419A-A2BD-A0C807E6FA0E}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HelloWorld", "HelloWorld\HelloWorld.csproj", "{7AB16EE9-4A79-4BFB-8C0B-D1A076E34896}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HelloWorld", "HelloWorld\HelloWorld.csproj", "{7AB16EE9-4A79-4BFB-8C0B-D1A076E34896}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WeatherUpdateClient", "WeatherUpdater\WeatherUpdateClient\WeatherUpdateClient.csproj", "{37B320C0-9E44-44EF-B89A-E671DBA4986B}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "WeatherUpdateClient", "WeatherUpdater\WeatherUpdateClient\WeatherUpdateClient.csproj", "{37B320C0-9E44-44EF-B89A-E671DBA4986B}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WeatherUpdateServer", "WeatherUpdater\WeatherUpdateServer\WeatherUpdateServer.csproj", "{5B8F7740-EFD5-46B9-8A96-2E1F47385295}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "WeatherUpdateServer", "WeatherUpdater\WeatherUpdateServer\WeatherUpdateServer.csproj", "{5B8F7740-EFD5-46B9-8A96-2E1F47385295}"
```
Comment (drewnoakes) [inline src/Samples.sln:14]: These project type GUID changes are not necessary and should be reverted.

File: src/Titanic Pattern/TitanicUsageExample/TitanicWorkerExample/TitanicWorkerExample.csproj [hunk 17]
```diff
@@ -1,133 +1,39 @@
...
-    <AssemblyFileVersionSettings>
-    </AssemblyFileVersionSettings>
-    <AssemblyInfoVersionSettings>
-    </AssemblyInfoVersionSettings>
+    <AssemblyVersionSettings />
+    <AssemblyFileVersionSettings />
+    <AssemblyInfoVersionSettings />
+    <AssemblyTitle>MDPWorkerExample</AssemblyTitle>
+    <Product>MDPWorkerExample</Product>
+    <Copyright>Copyright ©  2015</Copyright>
```
Comment (drewnoakes) [inline src/Titanic Pattern/TitanicUsageExample/TitanicWorkerExample/TitanicWorkerExample.csproj]: All these can be deleted.

File: src/Beacon/BeaconDemo/Properties/AssemblyInfo.cs [hunk 18]
```diff
@@ -2,35 +2,10 @@
 using System.Runtime.CompilerServices;
```
Comment (drewnoakes) [inline src/Beacon/BeaconDemo/Properties/AssemblyInfo.cs]: All these `AssemblyInfo.cs` files can be deleted. These projects aren't using COM.

**Review by drewnoakes — COMMENTED** [review]
Nice work. Left some comments on further clean up that would be good to add, to tidying things even further.

I'm curious, did you convert these files by hand, or did you use a tool? If it was a tool, which did you use?

**Comment by nicolaspierre1990** [comment]
I lost track of this, I used project2015to2017.migrate2019.tool for the migration, I will fix the comments

File: src/Majordomo/MDPClientExample/MDPClientExampleProgram.cs (hunk 12 above)
Comment (nicolaspierre1990) [reply src/Majordomo/MDPClientExample/MDPClientExampleProgram.cs]: Probably when trying things, reverted

File: src/Majordomo/MDPWorkerExample/MDPWorkerExampleProgram.cs (hunk 13 above)
Comment (nicolaspierre1990) [reply src/Majordomo/MDPWorkerExample/MDPWorkerExampleProgram.cs]: See previous

File: src/Majordomo/MajordomoProtocolTests/MDPBrokerTests.cs (hunk 14 above)
Comment (nicolaspierre1990) [reply src/Majordomo/MajordomoProtocolTests/MDPBrokerTests.cs]: Removed in Nunit 3.0
https://docs.nunit.org/articles/nunit/release-notes/breaking-changes.html

File: src/Pirate Pattern/Lazy Pirate/LazyPirate.Client2/Program.cs (hunk 15 above)
Comment (nicolaspierre1990) [reply src/Pirate Pattern/Lazy Pirate/LazyPirate.Client2/Program.cs]: @drewnoakes: Enum was made private, any reason for this ?

**Comment by nicolaspierre1990** [comment]
still WIP

**Comment by drewnoakes** [comment]
@nicolaspierre1990 I pulled your branch locally to play with it, then went through and applied a bunch more modernisations to the code across the solution. Pushed those here, and will merge them in one hit.

**Comment by nicolaspierre1990** [comment]
Sorry I forgot about this PR :D Glad it's merged
